core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -287,12 +288,25 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
   def recordAndGetThrottleTimeMs(session: Session, clientId: String, value: Double, timeMs: Long): Int = {
     var throttleTimeMs = 0
     val clientSensors = getOrCreateQuotaSensors(session, clientId)
+    val clientMetric = metrics.metrics().get(clientRateMetricName(clientSensors.metricTags))
+
+    // Record bound first before QuotaViolationException, otherwise the metric might never be logged
```

Comment:
Minor: I feel this comment is not useful after the code change.

Reply:
Fixed